pftools/python/parflow/tools/builders.py

```diff
@@ -114,86 +115,113 @@ def write(self, name, xllcorner=0, yllcorner=0, cellsize=0, vtk=False, extra=Non
+                            else self.patch_ids_bottom[j, i]
+                        bottom_mask[j, i] = patch_value if patch_value \
+                            else self.bottom_id
+
+                        # Top (+z)
+                        patch_value = 0 if self.patch_ids_top is None \
+                            else self.patch_ids_top[j, i]
+                        top_mask[j, i] = patch_value if patch_value \
+                            else self.top_id
+
+            # Generate asc / sa files
+            write_func = write_patch_matrix_as_asc
+            settings = {
+                'xllcorner': xllcorner,
+                'yllcorner': yllcorner,
+                'cellsize': cellsize,
+                'NODATA_value': 0,
+            }
+            short_name = name[:-6]
 
-        left_file_path = get_absolute_path(f'{short_name}_left.asc')
-        write_func(left_mask, left_file_path, **settings)
+            left_file_path = get_absolute_path(f'{short_name}_left.asc')
+            write_func(left_mask, left_file_path, **settings)
 
-        right_file_path = get_absolute_path(f'{short_name}_right.asc')
-        write_func(right_mask, right_file_path, **settings)
+            right_file_path = get_absolute_path(f'{short_name}_right.asc')
+            write_func(right_mask, right_file_path, **settings)
 
-        front_file_path = get_absolute_path(f'{short_name}_front.asc')
-        write_func(front_mask, front_file_path, **settings)
+            front_file_path = get_absolute_path(f'{short_name}_front.asc')
+            write_func(front_mask, front_file_path, **settings)
 
-        back_file_path = get_absolute_path(f'{short_name}_back.asc')
-        write_func(back_mask, back_file_path, **settings)
+            back_file_path = get_absolute_path(f'{short_name}_back.asc')
+            write_func(back_mask, back_file_path, **settings)
 
-        top_file_path = get_absolute_path(f'{short_name}_top.asc')
-        write_func(top_mask, top_file_path, **settings)
+            top_file_path = get_absolute_path(f'{short_name}_top.asc')
+            write_func(top_mask, top_file_path, **settings)
 
-        bottom_file_path = get_absolute_path(f'{short_name}_bottom.asc')
-        write_func(bottom_mask, bottom_file_path, **settings)
+            bottom_file_path = get_absolute_path(f'{short_name}_bottom.asc')
+            write_func(bottom_mask, bottom_file_path, **settings)
+
+            args = [
+                f'--mask-top {top_file_path}',
+                f'--mask-bottom {bottom_file_path}',
+                f'--mask-left {left_file_path}',
+                f'--mask-right {right_file_path}',
+                f'--mask-front {front_file_path}',
+                f'--mask-back {back_file_path}',
+                f'--pfsol {output_file_path}'
+            ]
+
+        else:
+            temp_pfb_file = tempfile.NamedTemporaryFile(suffix='.pfb', delete=False)
```

Comment:
Yes - my bad. I mistakenly assumed that not saying `delete=False` will cause immediate deletion of the temporary file since I'm not using this as a context manager as is normally done (and closing it will finally delete it). I should have checked. Removed that argument now (closing the file further down does delete the file).